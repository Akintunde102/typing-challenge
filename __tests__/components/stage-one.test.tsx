import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import StageOne from '../../components/stage-one';

jest.mock("../../resources/text", () => ({
    texts: ["TextOne", "TextTwo", "TextThree"]
}))

describe('Stage One Component', () => {

    const mockSetText = jest.fn();
    const mockSetStage = jest.fn();

    describe("when presentStage is not set to previous stage", () => {
        let renderedComponent: any;

        beforeEach(() => {
            renderedComponent = render(<StageOne setText={mockSetText} setStage={mockSetStage} presentStage={1} />)
        })

        it('does not renders a text', () => {
            expect(renderedComponent.container).toBeEmptyDOMElement();
        })

        it('matches snapshot', () => {
            expect(renderedComponent).toMatchSnapshot()
        })

    })

    describe("when presentStage is set to previous stage", () => {
        let renderedComponent: any;

        beforeEach(() => {
            renderedComponent = render(<StageOne setText={mockSetText} setStage={mockSetStage} presentStage={0} />)
        })

        it('shows title to pick text', () => {
            const text = screen.getByText("Pick Text");
            expect(text).toBeInTheDocument()
        })


        it('shows list of texts', () => {
            const textOne = screen.getByText("TextOne");
            const textTwo = screen.getByText("TextTwo");
            const textThree = screen.getByText("TextThree");
            expect(textOne).toBeInTheDocument();
            expect(textTwo).toBeInTheDocument();
            expect(textThree).toBeInTheDocument();
        })

        it('matches snapshot', () => {
            expect(renderedComponent).toMatchSnapshot()
        });

    })
});





