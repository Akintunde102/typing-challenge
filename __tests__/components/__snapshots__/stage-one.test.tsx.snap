// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stage One Component when presentStage is not set to previous stage matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div />
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Stage One Component when presentStage is set to previous stage matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="main"
      >
        <h1
          class="title"
        >
          Pick Text
        </h1>
        <code
          class="code"
        >
          Pick the Text You want Below
        </code>
        <div
          class="tContainer"
        >
          <button>
            Choose Text
          </button>
          TextOne
        </div>
        <div
          class="tContainer"
        >
          <button>
            Choose Text
          </button>
          TextTwo
        </div>
        <div
          class="tContainer"
        >
          <button>
            Choose Text
          </button>
          TextThree
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="main"
    >
      <h1
        class="title"
      >
        Pick Text
      </h1>
      <code
        class="code"
      >
        Pick the Text You want Below
      </code>
      <div
        class="tContainer"
      >
        <button>
          Choose Text
        </button>
        TextOne
      </div>
      <div
        class="tContainer"
      >
        <button>
          Choose Text
        </button>
        TextTwo
      </div>
      <div
        class="tContainer"
      >
        <button>
          Choose Text
        </button>
        TextThree
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
